lib/controlled-component.js

```diff
@@ -0,0 +1,43 @@
+import Component from './component';
+import {EMPTY_DIV} from './dom-patcher';
+
+export default class ControlledComponent extends Component {
+  createdCallback() {
+    super.createdCallback(...arguments);
+    this.controller = this.getConfig(`controller`);
+    if (!this.controller) {
+      throw Error(`"controller" must be set in config of a ControlledComponent`);
+    }
+    // Don't allow component's update directly
+    this._update = this.update;
+    this.update = () => { throw new Error(`update() not allowed from component. Use controller`); };
+    this._unsubscribeUpdates = this.controller.subscribeUpdates(() => this._update());
+  }
+
+  detachedCallback() {
+    super.detachedCallback();
+    this._unsubscribeUpdates();
```

Comment:
Is this really so much better than
```
this.updateListener = () => this._update();
this.controller.subscribeUpdates(this.updateListener);
...
this.controller.unsubscribeUpdates(this.updateListener);
```
Can't help it, sorry :) I guess it's minor, but this just seems clearer to me, esp if this is a publicly exposed mechanism